pygments/lexers/q.py

```diff
@@ -0,0 +1,228 @@
+
+    Lexer for the Q programming language.
+
+    :copyright: Copyright 2006-2022 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+from pygments.lexer import RegexLexer, words, include, bygroups
+from pygments.token import (
+    Comment,
+    Keyword,
+    Name,
+    Number,
+    Operator,
+    Punctuation,
+    String,
+    Text,
+    Whitespace,
+    Literal,
+    Generic,
+)
+
+__all__ = ["KLexer", "QLexer"]
+
+
+class KLexer(RegexLexer):
+    """
+    For `K <http://kx.com/>`_ source code.
+
+    .. versionadded:: 2.12
+    """
+
+    name = "K"
+    aliases = ["k"]
+    filenames = ["*.k"]
+
+    tokens = {
+        "whitespace": [
+            # hashbang script
+            (r"^#!.*$", Comment.Hashbang),
+            # Comments
+            (r"^/\s*\n", Comment.Multiline, "comments"),
+            (r"[ \t]+/.*", Comment.Single),
+            (r"^/.*\S.*", Comment.Single),
+            # Whitespace
+            (r"\s+", Whitespace),
+            # Strings
+            (r"\"", String.Double, "strings"),
+        ],
+        "root": [
+            include("whitespace"),
+            include("keywords"),
+            include("declarations"),
+        ],
+        "keywords": [
+            (
+                words(
+                    ("abs", "acos", "asin", "atan", "avg", "bin", "binr", "by")
+                    + ("cor", "cos", "cov", "dev", "delete", "div", "do")
+                    + ("enlist", "exec", "exit", "exp", "from", "getenv", "hopen")
+                    + ("if", "in", "insert", "last", "like", "log", "max", "min", "prd")
+                    + ("select", "setenv", "sin", "sqrt", "ss", "sum")
+                    + ("tan", "update", "var")
+                    + ("wavg", "while", "within", "wsum", "xexp"),
+                    suffix=r"\b",
+                ),
+                Keyword,
+            ),
+        ],
+        "declarations": [
+            (r"^\\t[s]?", Comment.Preproc),
+            (
+                r"^(\\\w\s+[^/\n]*?)([/].*)",
+                bygroups(Comment.Preproc, Comment.Single),
+            ),
+            (r"^\\\w+.*$", Comment.Preproc),
+            # Prompt
+            (r"^[a-zA-Z]\)", Generic.Prompt),
+            # Function Names
+            (
+                r"([.]?[a-zA-Z](?:\w|[.])*)(\s*)([-.~=!@#$%^&*_+|,<>?/\\:']?:)\s*(\{)",
+                bygroups(Name.Function, Whitespace, Operator, Punctuation),
+                "functions",
+            ),
+            # Variable Names
+            (
+                r"([.]?[a-zA-Z](?:\w|[.])*)(\s*)([-.~=!@#$%^&*_+|,<>?/\\:']?:)",
+                bygroups(Name.Variable, Whitespace, Operator),
+            ),
+            # Functions
+            (r"\{", Punctuation, "functions"),
+            # Parentheses
+            (r"\(", Punctuation, "parentheses"),
+            # Brackets
+            (r"\[", Punctuation, "brackets"),
+            # Errors
+            (r"'`[a-zA-Z]*(\w|[.])*", Name.Exception),
```

Comment:
Do you really mean the star on `[a-zA-Z]`? (Same below.)

As a mundane matter of style, `[\w.]` is a little easier to read than `(\w|[.])`.

Reply:
yes, i do require the `*` after `[a-zA-Z]` because the null symbol `` ` `` is a valid symbol.

thank you for the `[\w.]` suggestion. i didn't realize you could put `\w` within `[]`